### ISSUE TITLE
huge gaps in z12, visually missing streets

### ISSUE THREAD

**Issue by dieterdreist** [gravitystorm/openstreetmap-carto#3479]
### Expected behavior
consistent map browsing experience 
### Actual behavior
misleading voids because of missing minor roads in z12
### Links and screenshots illustrating the problem
https://www.openstreetmap.org/#map=12/41.8970/12.4837

![screenshot_2018-10-26 openstreetmap](https://user-images.githubusercontent.com/5439713/47560645-1ab0bc00-d919-11e8-852b-f60a0a290302.png)

**Comment by matkoniecz** [comment]
Can you mark areas where this is a problem?

**Comment by matthijsmelissen** [comment]
I don't think these voids are misleading, as these are areas where you cannot get by car. I think car navigation is more important than showing the city structure at that zoom level.

**Comment by dieterdreist** [comment]
sent from a phone

> On 25. Oct 2018, at 22:56, Matthijs Melissen <notifications@github.com> wrote:
> 
> I don't think these voids are misleading, as these are areas where you cannot get by car. I think car navigation is more important than showing the city structure at that zoom level
> 


then it should look again very different and you should render specific restrictions, you cannot go on the visible roads nearby either, unless you are a resident or otherwise authorized (limited traffic zone). The whole center is restricted.

https://romamobilita.it/it/servizi/ztl/ztl-centro-storico

**Comment by matkoniecz** [comment]
@dieterdreist Can you mark areas where this is a problem? I am unsure what parts of rendering are problematic (I can guess but I prefer to be sure before spending time describing my reasoning for this particular rendering decisions).

**Comment by dieterdreist** [comment]
Here you go:
![screen shot 2018-10-26 at 09 51 55](https://user-images.githubusercontent.com/581288/47552622-6bb6b500-d905-11e8-8aa1-ad22816a11df.png)

**Comment by Adamant36** [comment]
I hate that. Northern California has the same problem at Z7.

**Comment by matkoniecz** [comment]
> I hate that. Northern California has the same problem at Z7.

Can you give an example of affected location?

**Comment by matkoniecz** [comment]
In case of z12 I made assumed that

- landuse=residential is passable for pedestrians
- landuse=residential is passable for local vehicle traffic

I also wanted to avoid problem of showing too much data that is not only not giving more information but rather obscures it - see https://www.openstreetmap.org/user/Mateusz%20Konieczny/diary/34957 for this problem in the old map style.

At the same time I considered desirable to show major roads.

So in the end I decided to rely on landuse tagging at z12 to show that area is passable for local traffic and pedestrians and do not show minor roads there. It is possible to show minor roads at this zoom level but it makes impossible to distinguish structure of roads of a higher range.

Note that areas you marked are exactly areas without landuse tagging.

Note that I considered "landuse is no longer completely obscured by roads at z12 and similar" as not a problem.

Also, showing minor roads at z12 would make impossible to keep them from forming blob where individual roads are not distinguishable in cities, even for people with perfect monitors and without any eyesight problems. 

------

To summarize: it works as expected, with tradeoffs as expected - road network is visible in cities, missing landuse is visible and leads to worse rendering (what, given that one of this map style goals is to give feedback to mappers I considered as acceptable and maybe even desirable).

So: it works like expected but different priorities would lead to different rendering.

**Comment by dieterdreist** [comment]
Am Fr., 26. Okt. 2018 um 10:48 Uhr schrieb Mateusz Konieczny <
notifications@github.com>:

> In case of z12 I made assumed that
>
>    - landuse=residential is passable for pedestrians
>    - landuse=residential is passable for local vehicle traffic
>
>

IMHO landuse does not have implications on accessibility by foot or
vehicle, at least not "residential" landuse and as it is often used
(including roads). If it were only used in a detailed mapping way,
landuse=residential/commercial/retail/industrial etc. would not contain
public roads and residential landuse could probably be seen as likely
accessible only to residents (in most cases), although there are also
"semi-public" areas in many urban contexts, where ownership is private but
accessibility is permissive.

What bothers me with current z12 rendering is that it hides pedestrian
roads, which are roads just as are residential roads, with the exception
that general vehicle traffic is excluded. Not showing them here
misrepresents the urban structure.

The other problem is the absence of buildings, which at this zoom level are
already essential in many areas to better understand the space.

**Comment by matkoniecz** [comment]
> IMHO landuse does not have implications on accessibility by foot or vehicle, at least not "residential" landuse and as it is often used

I fully agree. But I assume that areas of landuse=residential/commercial/retail large enough to be noticeable at z12 are passable for local residents,

> What bothers me with current z12 rendering is that it hides pedestrian roads

I am pretty sure that it was already reported and discussed.

> The other problem is the absence of buildings, which at this zoom level are already essential in many areas to better understand the space.

buildings at this zoom level are noise and useful only in cases where landuse is not mapped

**Comment by mboeringa** [comment]
> The other problem is the absence of buildings, which at this zoom level are already essential in many areas to better understand the space.

> buildings at this zoom level are noise and useful only in cases where landuse is not mapped

I have to agree with @matkoniecz here, Z12 is something like a 1:100k-1:200k scale map. Showing individual buildings at that scale is nearly impossible, and will certainly not add a lot besides ultra dense clutter. Almost all official topographic maps start showing individual buildings only from scale 1:50k downwards, and even at that 1:50k scale, usually in generalized form for dense urban areas like a city center. Only 1:25k scale maps generally show mostly individual buildings or building blocks (city centers).

Buildings at Z12 is stretching it...

**Comment by kocio-pl** [comment]
I also think that z12 should not show every possible street type - it's enough to know what type of area is here to not make a high frequency noise, just to see a road backbone.

**Comment by matthijsmelissen** [comment]
As there seems to be no support for this proposal, I'm going to close this issue/

**Comment by matkoniecz** [comment]
> there are other places with similar problems in the centers, e.g. https://www.openstreetmap.org/#map=12/31.6287/-7.9889

Can you mark problematic areas that are not missing landuse?

**Comment by matkoniecz** [comment]
All areas marked in https://github.com/gravitystorm/openstreetmap-carto/issues/3479#issuecomment-433321407 are generally without landuse=residential or any other standard landuse=* and generally not covered by areas like tourism=museum or areas like hospital, park, university, school or others that you mentioned.

**Comment by matkoniecz** [comment]
There may be small areas of landuse=* or other mentioned above but they cover less than 10% of this area.

**Comment by dieterdreist** [comment]
sent from a phone

> On 28. Oct 2018, at 06:33, Mateusz Konieczny <notifications@github.com> wrote:
> 
> There may be small areas of landuse=* or other mentioned above but they cover less than 10% of this area.


there are 3 areas marked, of which 2 are entirely consisting of the mentioned features. For centres there is generally no appropriate landuse in OSM.

**Comment by matkoniecz** [comment]
I checked https://overpass-turbo.eu/s/Daj

From what I see 10% is an underestimate but 100% is also an overestimate.

**Comment by dieterdreist** [comment]
sent from a phone

> On 28. Oct 2018, at 08:41, Mateusz Konieczny <notifications@github.com> wrote:
> 
> From what I see 10% is an underestimate but 100% is also an overestimate.


I see gaping holes in the center and believe it is a misrepresentation. Not showing military landuse any more at z12 also contributes to this effect.

**Comment by matkoniecz** [comment]
Only small military area are hidden at z12.

Showing them as built up areas would not work as many are rural or wild areas. 

> I see gaping holes in the center and believe it is a misrepresentation. 

About 20% of area is mapped with tourism, leisure, amenity areas and not mapped with landuse areas (though I suspect that landuse=commercial may be suitable).

None of that is rendered as built up area on z12 leading to complete gap.

One large area fully covered by amenity/tourism/leisure areas, not covered by landuse is found my plan is to.

(1) all tagging community whatever they sure sure that no landuse value is applicable, including landuse=commercial

(2) in case of confirmation - add rendering of built up area on z12 and similar for this areas

**Comment by dieterdreist** [comment]
sent from a phone

> On 28. Oct 2018, at 10:59, Mateusz Konieczny <notifications@github.com> wrote:
> 
> About 20% of area is mapped with tourism, leisure, amenity areas and not mapped with landuse areas (though I suspect that landuse=commercial may be suitable).


it is a mix of commercial, retail, tourism/culture, residential and administrative that is typical for city centers, plus a lot of archaeological sites. Are theatres commercial landuse? Museums? Restaurants? Hotels?

**Comment by matkoniecz** [comment]
I think that most of them fit, due to significant implications I probably should confirm it on tagging mailing list.

**Comment by matkoniecz** [comment]
I posted message on tagging mailing list about ones that seem to be obviously landuse=commercial.

Cultural ones like archaeological sites and museums or theatres are a bit trickier and I will keep them for the next thread.

**Comment by jeisenbe** [comment]
Restaurants are retail, hotels are usually commercial, but they can be
retail if the ground level has a restaurant and shop.
